cpp/src/slice2matlab/Main.cpp

```diff
@@ -29,3675 +26,314 @@
-                }
-                break;
-            }
-            case Builtin::KindShort:
-            {
-                if (optional)
-                {
-                    out << nl << v << " = " << stream << ".readShortOpt(" << tag << ");";
-                }
-                else
-                {
-                    out << nl << v << " = " << stream << ".readShort();";
-                }
-                break;
-            }
-            case Builtin::KindInt:
-            {
-                if (optional)
-                {
-                    out << nl << v << " = " << stream << ".readIntOpt(" << tag << ");";
-                }
-                else
-                {
-                    out << nl << v << " = " << stream << ".readInt();";
-                }
-                break;
-            }
-            case Builtin::KindLong:
-            {
-                if (optional)
-                {
-                    out << nl << v << " = " << stream << ".readLongOpt(" << tag << ");";
-                }
-                else
-                {
-                    out << nl << v << " = " << stream << ".readLong();";
-                }
-                break;
-            }
-            case Builtin::KindFloat:
-            {
-                if (optional)
-                {
-                    out << nl << v << " = " << stream << ".readFloatOpt(" << tag << ");";
-                }
-                else
-                {
-                    out << nl << v << " = " << stream << ".readFloat();";
-                }
-                break;
-            }
-            case Builtin::KindDouble:
-            {
-                if (optional)
-                {
-                    out << nl << v << " = " << stream << ".readDoubleOpt(" << tag << ");";
-                }
-                else
-                {
-                    out << nl << v << " = " << stream << ".readDouble();";
-                }
-                break;
-            }
-            case Builtin::KindString:
-            {
-                if (optional)
-                {
-                    out << nl << v << " = " << stream << ".readStringOpt(" << tag << ");";
-                }
-                else
-                {
-                    out << nl << v << " = " << stream << ".readString();";
-                }
-                break;
-            }
-            case Builtin::KindObject:
-            case Builtin::KindValue:
-            {
-                assert(!optional); // Optional classes are disallowed by the parser.
-                out << nl << stream << ".readValue(" << v << ", 'Ice.Value');";
-                break;
-            }
-            case Builtin::KindObjectProxy:
             {
-                if (optional)
-                {
-                    out << nl << v << " = " << stream << ".readProxyOpt(" << tag << ");";
-                }
-                else
+                lock_guard lock(globalMutex);
```

Comment:
I'm not following this. Why do we need to exit with a failure if we receive a signal.